doc/Mellanox-npapi-thermal-psu-design

```diff
@@ -0,0 +1,132 @@
+Thermal design
+
+Terms
+thermal category: stands for the device category the thermal monitors for, including cpu core, cpu pack, asic, module and psu.
+switch model: stands for a set of SKUs that share the same PCB, port portfolios
+
+The purpose of thermal design is to represent:
+    1. the logic in which data in thermal sensor, which is required by each api, are accessed. In other words, how the set of APIs defined in thermal_base are implemented.
+    2. how thermals are managed by chassis, that is, how the diversity of thermal hierarchy regarding SKU be handled.
+
+5 apis are defined in thermal_base.py. get_temperature and get_high_threshold will be supported and detailed below. The other three interfaces will not be implemented for all categories of devices due to the following reason:
+1. get_low_threshold and set_low_threshold, currently we don't have interface regarding that. For SONiC which is mainly deployed in data center, low temperature threshold is less likely to be a concern. 
+2. set_high_threshold, currently it's read only. I think each device's high threshold, exceeding which a cool action should be triggered, should be designated by the device vendor since configuring it to a very high value might damage the device.
+
+Typically, a file in /var/run/hw-management/thermal represents the data each api required for each thermal device. Filename conventions regarding device categories and SKUs are similar but there is still minor difference that should be well handled, like:
+    1. All SKUs can have thermals in different type of devices, including cpu core, cpu pack, asic and each sfp modules.
+    2. The filename conventions containing data for each API differ on a thermal category basis.
+    3. Different SKUs have different number for each category of devices, like number of cpu cores, sfp modules, therefore their thermals differing.
+
+The files that contain data for each api/thermal category are listed below (n is index of the device):
+1. cpu_core:
+    cpu_core<n>, contains the temperature of the core, for api "get_temperature".
+    cpu_core_alarm<n>, contains the threshold when temperature exceeds some cooling action is recommended. it can be used for api "get_high_threshold".
+2. cpu_pack: the files prefixed with cpu_pack have the same meaning as those with cpu_core.
+3. sfp module:
+    temp_input_module<n>, contains the temperature of the sfp module, for api "get_temperature"
+    temp_crit_module<n>, contains the critical threshold, for api "get_high_threshold"
+4. psu module:
+    psu<n>, contains the temperature of the psu module, for api "get_temperature"
+    no support for "get_high_threshold"
```

Comment:
IMO there is a high_threshold for PSU temp

Reply:
fixed.